src/app/admin/websiteDataTable/CustomEditComponent.tsx

```diff
@@ -0,0 +1,57 @@
+import { GridRenderEditCellParams, useGridApiContext } from "@mui/x-data-grid";
+import React, { useEffect, useLayoutEffect, useRef, useState } from "react";
+
+interface CustomCellProps extends GridRenderEditCellParams {
+    editMode: boolean;
+}
+
+const CustomComponent: React.FC<CustomCellProps> = (props: CustomCellProps) => {
+    const { id, field, hasFocus } = props;
+    const apiRef = useGridApiContext();
+    const ref = useRef<HTMLTextAreaElement>(null);
+    const initialValue = useRef(props.value);
+    const [value, setValue] = useState<string>(props.value);
+
+    useEffect(() => {
+        initialValue.current = props.value;
+        setValue(props.value);
+    }, [props.value]);
+
+    useLayoutEffect(() => {
+        if (hasFocus) {
```

Comment:
We could additionally check for ref.current since everything inside the if is to do with ref.current anyway
```suggestion
        if (hasFocus && ref.current) {
```